### PR TITLE
New distributive rpm

### LINKED ISSUE #633: Common health checks
We need health checks for more basic, MI-unrelated things. I'd like to see the following on every node:

 * CPU usage (<100%)
 * Disk usage (<90%)
 * Memory usage (<100%)
 * ~~`centos` user present~~

Does anybody else have any ideas?

### PR DESCRIPTION
**Pull request by siddharthist**
Fixes #633, #717, #740, enables #1045.
We will need to switch to CiscoCloud Bintray before merging.
Tested on Vagrant.

### CODE REVIEW

**Comment by TanyaCouture** [comment]
Successfully installed on:
Vagrant
AWS

**Comment by TanyaCouture** [comment]
Openstack installation:
The UI display says:
    Checks: Unknown
        No Health Checks loaded

File: roles/common/defaults/main.yml [hunk 1]
```diff
@@ -2,4 +2,5 @@
 use_host_domain: false
 host_domain: novalocal
 selinux_policy: targeted
-selinux_state: permissive
\ No newline at end of file
+selinux_state: permissive
+distributive_rpm_url: "https://bintray.com/artifact/download/siddharthist/distributive-testing/distributive-0.2.4-1.x86_64.rpm"
```
Comment (ryane) [inline roles/common/defaults/main.yml]: will we be able to get this on an "official" bintray soon?

Comment (siddharthist) [inline roles/common/defaults/main.yml]: I would love it if anyone on our team with access could just download it from there and reupload it to CiscoCloud.

**Comment by ryane** [comment]
With this version, I am seeing failing checks on all control nodes for the following two services:

distributive-marathon-checks

```
time="2016-01-22T01:25:13Z" level=info msg="Creating checklist(s)..." path="/etc/distributive.d/marathon.json" type=file 
time="2016-01-22T01:25:13Z" level=fatal msg="Invalid check ID" id= 
```

distributive-mantlui-checks

```
time="2016-01-22T01:18:35Z" level=info msg="Creating checklist(s)..." path="/etc/distributive.d/mantlui.json" type=file 
time="2016-01-22T01:18:35Z" level=info msg="Running check File" 
time="2016-01-22T01:18:35Z" level=info msg="Running check DockerImageRegexp" 
time="2016-01-22T01:18:35Z" level=info msg="Running check PortTCP" 
time="2016-01-22T01:18:35Z" level=info msg="Running check PortTCP" 
time="2016-01-22T01:18:35Z" level=info msg="Running check File" 
time="2016-01-22T01:18:35Z" level=info msg="Running check File" 
time="2016-01-22T01:18:35Z" level=info msg="Running check File" 
time="2016-01-22T01:18:35Z" level=info msg="Report from checklist" checklist="mantlui health checks" report="↴\nTotal: 7\nPassed: 6\nFailed: 1\nOther: 0\nNo such file or directory: /etc/mantlui.nginx"
```

(/etc/mantlui.nginx no longer exists as of #955)

**Comment by ryane** [comment]
also, once this is ready, should we just move it to the asteris bintray? A lot of our packages live there.

**Comment by siddharthist** [comment]
Thanks for testing @ryane, I'll update the package and this PR.

**Comment by siddharthist** [comment]
Rebased onto master, removed all marathon checks in preparation for adding them to the Marathon package. Works for me on Vagrant, needs further testing.

**Comment by siddharthist** [comment]
Rebased and squashed, ready for testing! Works for me on Vagrant.

**Comment by ryane** [comment]
I am getting this failure on all nodes (aws):

```
time="2016-02-01T21:34:22Z" level=info msg="Creating checklist(s)..." path="/etc/distributive.d/common.json" type=file 
time="2016-02-01T21:34:22Z" level=info msg="Running check Symlink" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Installed" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Installed" 
time="2016-02-01T21:34:22Z" level=info msg="Running check SystemctlUnitFileStatus" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Installed" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Installed" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Directory" 
time="2016-02-01T21:34:22Z" level=info msg="Running check File" 
time="2016-02-01T21:34:22Z" level=info msg="Running check CPUUsage" 
time="2016-02-01T21:34:22Z" level=info msg="Running check MemoryUsage" 
time="2016-02-01T21:34:22Z" level=info msg="Running check DiskUsage" 
time="2016-02-01T21:34:25Z" level=info msg="Report from checklist" checklist="common health check" report="↴\nTotal: 11\nPassed: 10\nFailed: 1\nOther: 0\nUnit file could not be found: firewalld.service"
```

**Comment by siddharthist** [comment]
I think I see the problem. I'll try again. Thanks once again @ryane!

**Comment by TanyaCouture** [comment]
Successful installation on:
Vagrant

**Comment by siddharthist** [comment]
Tested, working on Openstack

**Comment by ryane** [comment]
+1 on aws

**Pull request merged**